braid/braid_status.h

```diff
@@ -579,6 +596,16 @@ ACCESSOR_HEADER_GET1(Sync, CallingFunction,  Int)
 ACCESSOR_HEADER_GET1(Sync, NumErrorEst,      Int)
 ACCESSOR_HEADER_GET1(Sync, AllErrorEst,      Real)
 
```

Comment:
@jcc242 
I guess the same question is here, can we use the ACCESSOR_HEADER style to declare SyncStatusGetTComm ? I've never tried this for a non-built-in type like MPI_Comm, but its worth exploring, I think.

Reply:
@jbschroder We can make it use the ACCESSOR_HEADER and similarly for the implementation. I specifically chose not to because I wanted to guard access to the temporal communicator. That is, I didn't want to make it easy for someone to add access to temporal communicator to the other status functions and structs.

If you think it would be better to be consistent (or think guarding the temporal communicator in this manner is unnecessary), then I will use the ACESSOR_HEADER/FUNCTION methods.

Reply:
@rfalgout What do you think Rob?  I probably have a slight preference for consistency, i.e., using the ACCESSOR_HEADER/FUNCTION approach -- but I'm flexible, too.

Reply:
Hi @jcc242 and @jbschroder .  Sorry for the delayed response.  Isn't the goal of Sync to give the user access to the temporal communicator to do (essentially arbitrary) work in a temporally-synchronized manner?  If that is indeed the goal, then I think it's perfectly fine to give them access to the communicator through a Sync accessor function/macro, and that would be consistent with how we are currently doing things elsewhere (which admittedly isn't perfect).  So, I would vote for consistency as well, assuming I understand the issue correctly here.  We can prevent access in other status functions via the PR process.  Thanks!

Reply:
@rfalgout @jbschroder Hi! I have updated everything to fit your comments. I know use the ACCESSOR_HEADER/FUNCTION approach for GetTComm. I did have to create a new macro for this: ACCESSOR_FUNCTION_GET1_GENERIC

The reason for this new macro is that the standard ACCESSOR_FUNCTION_GET1 prepends the argument type with "braid_", but we don't have a braid_MPI_Comm type. This new macro does not prepend the argument types. Do you think I should `typedef MPI_Comm braid_MPI_Comm` instead of creating this new macro?

Thank you!

Reply:
I think you should add the typedef in `braid_defs.h` near the other MPI-related stuff.  Thanks!

Reply:
@rfalgout I have done so now.